### PR TITLE
feat(apps): add password hasher app

### PR DESCRIPTION
**Pull request by vdice**
- Adds a password hasher app with toggles for cpu (iterations), mem usage, password value and sleep

### CODE REVIEW

File: apps/password-hasher/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+    let mut mem_kib: u32 = 1000;
+
+    let parsed_url = url::Url::parse(req.uri()).unwrap();
+    let hash_query: std::collections::HashMap<_, _> =
+        parsed_url.query_pairs().into_owned().collect();
+
+    for (key, value) in hash_query.into_iter() {
+        match &*key {
+            // sleep=<ms> param simulates generic i/o time
+            "sleep" => {
```
Comment (vdice) [inline apps/password-hasher/src/lib.rs:22]: The sleep toggle originates from the beginning of this app which was derived from https://github.com/fermyon/spin/blame/13a133fded7c48fa5d1931207619379e737e9376/crates/trigger-http/benches/spin-http-benchmark/src/lib.rs.  Is this too kitchen-sinky?  Do we want a separate sleep app?

File: apps/password-hasher/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            // password=<password> param is the user-supplied password
+            "password" => {
+                password = value.parse::<String>().expect("invalid password string");
+            }
+            // cpu=<iterations> param increases compute time
+            "cpu" => {
+                iterations = value.parse().expect("invalid cpu iteration count");
+            }
+            // mem=<KiB> param increases memory usage
+            "mem" => {
```
Comment (vdice) [inline apps/password-hasher/src/lib.rs:35]: In its current form, CPU usage will still be heavy when mem usage is notched up, so this isn't a good method for a "mem-intensive-only".  We do have https://github.com/fermyon/spinkube-performance/tree/main/apps/mem-intensive or we can delete that one and update this app to try to isolate memory-bound behavior better.

File: apps/password-hasher/src/lib.rs (hunk 1 above)
Comment (lann) [reply apps/password-hasher/src/lib.rs:23]: Being kitchen-sinky seems like an _advantage_ to me for this sort of thing. You could maybe call this "benchmarking-load" or something to make that more clear?

Comment (lann) [reply apps/password-hasher/src/lib.rs:23]: Oh I would maybe split these into two routes though, e.g. `/sleep?ms=...` and `/hash?cpu=...&mem=...` or something?

Comment (kate-goldenring) [reply apps/password-hasher/src/lib.rs:23]: I see the value of a kitchen sink app and just a sleep one. We could start with this as is, and if we want an app/endpoint that is only I/O, we can add it.

**Review by kate-goldenring — APPROVED** [review]
LGTM!

**Comment by vdice** [comment]
Ok, sounds like we're fine to get this in as-is and tweak/tune as needed.

**Pull request merged**